### ISSUE TITLE
`projects_iam` module rejects bindings for custom roles

### ISSUE THREAD

**Issue by bmckerry** [terraform-google-modules/terraform-google-iam#219]
### TL;DR

When using the `projects_iam` module to non-authoritatively bind an entity to a custom role, the module returns a 400 Bad Request error. We're using this module through terragrunt.

### Expected behavior

I expected the entity to be non-authoritatively bound to the custom role (which works when using `google_project_iam_member` directly).

### Observed behavior

The binding isn't created, and this error is given:
```hcl
╷
│ Error: Request `Create IAM Members roles/custom.vertex.prediction serviceAccount:{account}@{gcp_project}.iam.gserviceaccount.com for project "gcp_project"` returned error: Error applying IAM policy for project "gcp_project": Error setting IAM policy for project "gcp_project": googleapi: Error 400: Role roles/custom.vertex.prediction is not supported for this resource., badRequest
│
│   with google_project_iam_member.project_iam_additive["default--roles/custom.vertex.prediction--serviceAccount:{account}@{gcp_project}.iam.gserviceaccount.com"],
│   on main.tf line 51, in resource "google_project_iam_member" "project_iam_additive":
│   51: resource "google_project_iam_member" "project_iam_additive" {
│
╵
```

### Terraform Configuration

```hcl
locals {
  project = "gcp_project"
}

inputs = {
  projects = [
    local.project
  ]

  bindings = {
    "roles/custom.vertex.prediction" = [
      "serviceAccount:{account_name}@${local.project}.iam.gserviceaccount.com"
    ]
  }
}
```

### Terraform Version

```sh
% tf version
Terraform v1.5.3
on darwin_amd64
+ provider registry.terraform.io/hashicorp/google v4.66.0
+ provider registry.terraform.io/hashicorp/google-beta v4.66.0
+ provider registry.terraform.io/hashicorp/random v3.1.3
```

### Additional information

The role `roles/custom.vertex.prediction` exists in our GCP project's custom roles. I was able to assign the service account to the role through the GCP frontend console. We've seen this for any custom roles we've tried assigning via this module.

**Comment by bmckerry** [comment]
Hello, is there any update on this? We'd like to use this module, but not being able to use custom roles with it greatly hinders our usage.
